pkg/kubectl/service_basic.go

```diff
@@ -225,12 +226,25 @@ func (s ServiceCommonGeneratorV1) StructuredGenerate() (runtime.Object, error) {
 		})
 	}
 
-	// setup default label and selector
+	// setup default label
 	labels := map[string]string{}
 	labels["app"] = s.Name
-	selector := map[string]string{}
-	selector["app"] = s.Name
-
+	// setup selector according to selector flag, use default if nothing is provided
+	// selector will not be added for service type ExternalName
+	isTypeExternalName := s.Type == api.ServiceTypeExternalName
+	selectorString := s.Selector
+	var selector map[string]string
+	if !isTypeExternalName {
+		if len(selectorString) == 0 {
```

Comment:
so if we explicit a selector, the default selector won't be added ? I'm not sure if this is right ?
I thought it would be something like this
```
	// setup default label and selector
	labels := map[string]string{}
	labels["app"] = s.Name
	selector := map[string]string{}
	selector["app"] = s.Name

	if s.Type != api.ServiceTypeExternalName && len(s.Selector) {
		mergeSelector, err := ParseLabels(s.Selector)
		if err != nil {
			return nil, err
		}
		for k, v := range mergeSelector {
			selector[k] = v
		}
	}
```
correct me if I'm wrong